modules/objdetect/include/opencv2/objdetect/aruco_board.hpp

```diff
@@ -106,48 +81,55 @@ class CV_EXPORTS_W GridBoard : public Board {
      *
      * This function return the image of the GridBoard, ready to be printed.
      */
-    CV_WRAP void generateImage(Size outSize, OutputArray img, int marginSize = 0, int borderBits = 1) const CV_OVERRIDE;
+    CV_WRAP void generateImage(Size outSize, OutputArray img, int marginSize = 0, int borderBits = 1) const;
+
+    CV_DEPRECATED_EXTERNAL  // avoid using in C++ code, will be moved to “protected” (need to fix bindings first)
+    Board();
 
+protected:
+    Board(Ptr<BoardImpl> impl);
+    template<typename T> T& as() { CV_Assert(boardImpl); return static_cast<T&>(*boardImpl); }
+    template<typename T> const T& as() const { CV_Assert(boardImpl); return static_cast<const T&>(*boardImpl); }
+    Ptr<BoardImpl> boardImpl;
```

Comment:
> boardImpl

just `impl` as we should not expect similar mistakes with inherited multiple implementation fields